src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseSchedule.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.helper.builders
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseActivity
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.CourseSchedule
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.SlotCategory
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.CourseScheduleRepository
+import java.time.LocalDate
+import java.time.LocalTime
+
+@DslMarker
+annotation class CourseScheduleDslMarker
+
+@TestDataDslMarker
+interface CourseScheduleDsl
+
+@Component
+class CourseScheduleBuilderRepository(val courseScheduleRepository: CourseScheduleRepository) {
+  fun save(courseSchedule: CourseSchedule) = courseScheduleRepository.save(courseSchedule)
+}
+
+@Component
+class CourseScheduleBuilderFactory(val repository: CourseScheduleBuilderRepository? = null) {
+  fun builder() = CourseScheduleBuilder(repository)
+}
```

Comment:
CourseSchedules don't have any children so all the builder factory needs to know is how to save itself in the repository.